controllers/kubeadmcontrolplane_controller.go

```diff
@@ -161,6 +182,12 @@ func (r *KubeadmControlPlaneReconciler) reconcile(ctx context.Context, log logr.
 		return ctrl.Result{}, fmt.Errorf("deleting ControlPlaneUpgrade object: %v", err)
 	}
 
+	log.Info("Resuming control plane machine health check", "MachineHealthCheck", cpMachineHealthCheckName(kcp.Name))
+	delete(mhc.Annotations, clusterv1.PausedAnnotation)
+	if err := mhcPatchHelper.Patch(ctx, mhc); err != nil {
+		return ctrl.Result{}, fmt.Errorf("updating annotations for machine health check: %v", err)
+	}
```

Comment:
can you add this logic after line 127 as well? Maybe move this to a function and just call it at both places